solr/solrj-zookeeper/src/java/org/apache/solr/client/solrj/impl/MetricsFetcher.java

```diff
@@ -0,0 +1,202 @@
+package org.apache.solr.client.solrj.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.solr.client.solrj.response.SimpleSolrResponse;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.cloud.ZkStateReader;
+import org.apache.solr.common.params.CommonParams;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.common.util.StrUtils;
+import org.apache.solr.common.util.Utils;
+import org.apache.zookeeper.KeeperException;
+
+// uses metrics API to get node information
```

Comment:
Javadocs please.  If it was written for something in particular in mind, then it's very useful to say that, even if it may theoretically have future uses we don't know about.  It's not clear why would something use this instead of going directly to the Metrics API?  As a newcomer to this code, is it related to having short/simply well-known named "tags" (formerly maybe called snitch?) that loosely correlate to a more complex metrics expression, so that some other code can use these tags to make node placement decisions based on those same tags?

The lifecycle of an instance of this class is not obvious; please add a sentence on that.